From cb3f56b7fbadf2bdbd111117486e6ec966d68355 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Tue, 24 Jun 2025 14:55:36 -0400
Subject: [PATCH 149/166] fixup! Revert "winewayland: Use SEND_HWMSG_NO_RAW for
 keyboard."

This reverts commit 6fc63b9626c4b67819e71127ce94dc69463666f6.

This behavior is wrong
---
 dlls/winewayland.drv/wayland_keyboard.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dlls/winewayland.drv/wayland_keyboard.c b/dlls/winewayland.drv/wayland_keyboard.c
index 44d90749162..a26915c412c 100644
--- a/dlls/winewayland.drv/wayland_keyboard.c
+++ b/dlls/winewayland.drv/wayland_keyboard.c
@@ -814,7 +814,7 @@ static void send_right_control(HWND hwnd, uint32_t state)
     input.ki.wScan = 0xe000 | (key2scan(KEY_RIGHTCTRL) & 0xff);
     input.ki.dwFlags = KEYEVENTF_SCANCODE | KEYEVENTF_EXTENDEDKEY;
     if (state == WL_KEYBOARD_KEY_STATE_RELEASED) input.ki.dwFlags |= KEYEVENTF_KEYUP;
-    NtUserSendHardwareInput(hwnd, SEND_HWMSG_NO_RAW, &input, 0);
+    NtUserSendHardwareInput(hwnd, 0, &input, 0);
 }
 
 static void keyboard_handle_key(void *data, struct wl_keyboard *wl_keyboard,
@@ -840,7 +840,7 @@ static void keyboard_handle_key(void *data, struct wl_keyboard *wl_keyboard,
     if (scan & ~0xff) input.ki.dwFlags |= KEYEVENTF_EXTENDEDKEY;
 
     if (state == WL_KEYBOARD_KEY_STATE_RELEASED) input.ki.dwFlags |= KEYEVENTF_KEYUP;
-    NtUserSendHardwareInput(hwnd, SEND_HWMSG_NO_RAW, &input, 0);
+    NtUserSendHardwareInput(hwnd, 0, &input, 0);
 }
 
 static void keyboard_handle_modifiers(void *data, struct wl_keyboard *wl_keyboard,
-- 
2.50.0

